src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -322,5 +335,41 @@ ExpressionSyntax initializer
 				.GetSyntax()
 				.GetLocation();
 		}
+
+		private (
+			TypeSyntax TypeSyntax,
+			SyntaxToken Identifier,
+			ExpressionSyntax Initializer,
+			bool IsAutoImplemented,
+			bool IsReadOnly
+		) GetPropertySyntax( IPropertySymbol symbol, SyntaxNode syntax )
+			=> syntax switch {
+				PropertyDeclarationSyntax prop => (
+					prop.Type,
+					prop.Identifier,
+					prop.Initializer?.Value,
+					prop.IsAutoImplemented(),
+					symbol.IsReadOnly
+				),
+				ParameterSyntax param => (
+					param.Type,
+					param.Identifier,
+					Initializer: null,
+					IsAutoImplemented: true,
+
+					// The property X in "record Foo(int X);" is not IsReadOnly,
+					// but its SetMethod has IsInitOnly. We could probably just
+					// return true here but this seems safer.
+					IsReadOnly: symbol.SetMethod != null && symbol.SetMethod.IsInitOnly
+				),
+				_ => throw new NotImplementedException()
+			};
+
+		private static BaseListSyntax GetBaseList( SyntaxNode syntax )
+			=> syntax switch {
+				   ClassDeclarationSyntax @class => @class.BaseList,
+				   RecordDeclarationSyntax record => record.BaseList,
+				   _ => null
+			   };

```

Comment:
BaseList is `BaseTypeDeclarationSyntax`, needn't differentiate these two.

Reply:
This was to differentiate ClassDeclarationSyntax/RecordDeclarationSyntax not BaseList. It looks like they have a common supertype that has BaseList though (TypeDeclarationSyntax), I'll try that

Reply:
It worked